sdk/core/azure-core/src/main/java/com/azure/core/util/jsonpatch/JsonPatchDocument.java

```diff
@@ -89,18 +90,18 @@ public JsonPatchDocument appendAdd(String path, Object value) {
      *
      * <p><strong>Code Samples</strong></p>
      *
-     * {@codesnippet com.azure.core.experimental.jsonpatch.JsonPatchDocument.appendAddRaw#String-String}
+     * {@codesnippet com.azure.core.util.jsonpatch.JsonPatchDocument.appendAddRaw#String-String}
      *
      * @param path The path to apply the addition.
      * @param rawJson The raw JSON value that will be added to the path.
      * @return The updated JsonPatchDocument object.
      * @throws NullPointerException If {@code path} is null.
      */
     public JsonPatchDocument appendAddRaw(String path, String rawJson) {
-        return appendAddInternal(path, () -> Optional.ofNullable(rawJson));
+        return appendAddInternal(path, () -> Option.of(rawJson));
     }
 
-    private JsonPatchDocument appendAddInternal(String path, Supplier<Optional<String>> jsonSupplier) {
+    private JsonPatchDocument appendAddInternal(String path, Supplier<Option<String>> jsonSupplier) {
```

Comment:
wondering whether we really need a `Supplier` instance or just taking `Option` is good enough. We seem to invoke the supplier immediately.

Reply:
Removed, this could be brought back in the future to defer serialization of the Object until serialization or toString is called.